lib/trino-record-decoder/src/main/java/io/trino/decoder/RowDecoderFactory.java

```diff
@@ -13,10 +13,12 @@
  */
 package io.trino.decoder;
 
+import io.trino.spi.connector.ConnectorSession;
+
 import java.util.Map;
 import java.util.Set;
 
 public interface RowDecoderFactory
 {
-    RowDecoder create(Map<String, String> decoderParams, Set<DecoderColumnHandle> columns);
+    RowDecoder create(Map<String, String> decoderParams, Set<DecoderColumnHandle> columns, ConnectorSession session);
```

Comment:
TODO: extract to a separate commit if we are ok with this approach:
This is to support a common use case where an avro record has no fields.
The new behavior of avro 1.11.1 no longer allows missing fields so it has to be explicitly set by another decoder (EmptyFieldHandlingAvroDecoder).